src/team.cpp

```diff
@@ -243,17 +243,21 @@ void team::team_info::read(const config& cfg)
 		support_per_village = lexical_cast_default<int>(village_support, game_config::village_support);
 	}
 
-	controller = team::CONTROLLER::AI;
-	controller.parse(cfg["controller"].str());
+	std::optional<side_controller::TYPE> type = side_controller::get_enum(cfg["controller"].str());
```

Comment:
`optional::value_or` can be used here too.

Reply:
Also, while this can be replaced with a one-liner, I would recommend using `auto` instead of `std::optional<side_controller::TYPE>` other places where you need a variable.

Reply:
Done.